src/execution/operator/persistent/physical_export.cpp

```diff
@@ -159,18 +171,42 @@ SourceResultType PhysicalExport::GetData(ExecutionContext &context, DataChunk &c
 		return lhs.get().oid < rhs.get().oid;
 	});
 
-	// write the schema.sql file
+	catalog_entry_vector_t catalog_entries;
+	AddEntries(catalog_entries, schemas);
+	AddEntries(catalog_entries, custom_types);
+	AddEntries(catalog_entries, sequences);
+	AddEntries(catalog_entries, tables);
+	AddEntries(catalog_entries, views);
+	AddEntries(catalog_entries, indexes);
+	AddEntries(catalog_entries, macros);
+	return catalog_entries;
+}
+
+SourceResultType PhysicalExport::GetData(ExecutionContext &context, DataChunk &chunk,
+                                         OperatorSourceInput &input) const {
+	auto &state = input.global_state.Cast<ExportSourceState>();
+	if (state.finished) {
+		return SourceResultType::FINISHED;
+	}
+
+	auto &ccontext = context.client;
+	auto &fs = FileSystem::GetFileSystem(ccontext);
+
+	auto &catalog = Catalog::GetCatalog(ccontext, info->catalog);
+
 	// export order is SCHEMA -> SEQUENCE -> TABLE -> VIEW -> INDEX
+	catalog_entry_vector_t catalog_entries;
+	if (catalog.IsDuckCatalog()) {
+		auto &duck_catalog = catalog.Cast<DuckCatalog>();
+		auto &dependency_manager = duck_catalog.GetDependencyManager();
+		catalog_entries = GetDependencyDrivenExportOrder(context, dependency_manager, *info, exported_tables);
+	} else {
+		catalog_entries = GetNaiveExportOrder(context, *info, exported_tables);
```

Comment:
Sadly we can't get rid of this error-prone method entirely, though I haven't seen any other classes that derive from Catalog, but this could likely be a point of interest for extensions later on